src/ImageSharp.Drawing/Processing/Drawing/Brushes/GradientBrushes/LinearGradientBrush{TPixel}.cs

```diff
@@ -0,0 +1,176 @@
+            /// <param name="options">the graphics options</param>
+            public LinearGradientBrushApplicator(
+                ImageFrame<TPixel> source,
+                Point start,
+                Point end,
+                ColorStop<TPixel>[] colorStops,
+                GradientRepetitionMode repetitionMode,
+                GraphicsOptions options)
+                : base(source, options, colorStops, repetitionMode)
+            {
+                this.start = start;
+                this.end = end;
+
+                // the along vector:
+                this.alongX = this.end.X - this.start.X;
+                this.alongY = this.end.Y - this.start.Y;
+
+                // the cross vector:
+                this.acrossX = this.alongY;
+                this.acrossY = -this.alongX;
+
+                // some helpers:
+                this.alongsSquared = (this.alongX * this.alongX) + (this.alongY * this.alongY);
+                this.length = (float)Math.Sqrt(this.alongsSquared);
+            }
+
+            protected override float PositionOnGradient(int x, int y)
+            {
+                if (this.acrossX == 0)
+                {
+                    return (x - this.start.X) / (float)(this.end.X - this.start.X);
+                }
+                else if (this.acrossY == 0)
+                {
+                    return (y - this.start.Y) / (float)(this.end.Y - this.start.Y);
+                }
+                else
+                {
+                    float deltaX = x - this.start.X;
+                    float deltaY = y - this.start.Y;
+                    float k = ((this.alongY * deltaX) - (this.alongX * deltaY)) / this.alongsSquared;
+
+                    // point on the line:
+                    float x4 = x - (k * this.alongY);
+                    float y4 = y + (k * this.alongX);
+
+                    // get distance from (x4,y4) to start
+                    float distance = (float)Math.Sqrt(
+                        Math.Pow(x4 - this.start.X, 2)
+                        + Math.Pow(y4 - this.start.Y, 2));
+
+                    // get and return ratio
+                    float ratio = distance / this.length;
+                    return ratio;
+                }
+            }
+
+            public override void Dispose()
+            {
+            }
+
+            internal override void Apply(Span<float> scanline, int x, int y)
+            {
+                base.Apply(scanline, x, y);
+
+                // TODO: we should at least(!) speed up the x=0 and y=0 special cases.
+                // But in fact that could be done by special case Applicators directly:
+                // - horizontal would apply a precalc. row independent of given row,
+                // - vertical would get the color of the row once and fill the whole line.
+
+                // Span<TPixel> destinationRow = this.Target.GetPixelRowSpan(y).Slice(x, scanline.Length);
```

Comment:
can we drop the commented out code

Reply:
sure. dropped a note in the PR description here instead.